release.sh

```diff
@@ -48,19 +51,35 @@ if [ ! -z "${INPUT_LDFLAGS}" ]; then
     LDFLAGS_PREFIX="-ldflags"
 fi
 
+# fulfill GOAMD64 option
+if [ ! -z "${INPUT_GOAMD64}" ]; then
+    if [[ "${INPUT_GOARCH}" =~ amd64 ]]; then
+        GOAMD64_FLAG="${INPUT_GOAMD64}"
+    else
+        echo "GOAMD64 should only be use with amd64 arch." >>/dev/stderr
+        GOAMD64_FLAG=""
+    fi
+else
+    if [[ "${INPUT_GOARCH}" =~ amd64 ]]; then
+        GOAMD64_FLAG="v1"
+    else
+        GOAMD64_FLAG=""
+    fi
+fi
+
 # build
 BUILD_ARTIFACTS_FOLDER=build-artifacts-$(date +%s)
 mkdir -p ${INPUT_PROJECT_PATH}/${BUILD_ARTIFACTS_FOLDER}
 cd ${INPUT_PROJECT_PATH}
 if [[ "${INPUT_BUILD_COMMAND}" =~ ^make.* ]]; then
     # start with make, assumes using make to build golang binaries, execute it directly
-    GOOS=${INPUT_GOOS} GOARCH=${INPUT_GOARCH} eval ${INPUT_BUILD_COMMAND}
+    GOAMD64=${GOAMD64_FLAG} GOOS=${INPUT_GOOS} GOARCH=${INPUT_GOARCH} eval ${INPUT_BUILD_COMMAND}
```

Comment:
Will the `GOAMD64` be ignored prior to go1.18?  Maybe it's better to set `GOAMD64_FLAG='GOAMD64=${INPUT_GOAMD64}'`, and use `${GOAMD64_FLAG} GOOS=...` directly here.

Reply:
I confirmed it will be ignored. Since it will only be checked after Go 1.18.
https://github.com/golang/go/issues/45453